### DISCUSSION TITLE
Investigate how to integrate with django-forms-dynamic

### DISCUSSION THREAD

**Discussion by adamghill** [django-commons/django-unicorn#317] [Ideas]
https://github.com/dabapps/django-forms-dynamic looks like it might be useful for dynamically creating forms. It would be nice to see if it's possible to integrate with `Unicorn`.

**Comment by nerdoc** [comment]
It's not direct an answer here, but a question that has to do with that. I have a very dynamic form, but created it manually (in Form.__init__), where fields are created from a database (settings) model and their contents; basically, one model instance creates one field.
Works more or less, and I give each field a `unicorn:model` using
```python
field.widget.attrs["unicorn:model.lazy"] = fieldname
```
That works, Unicord does it's job - even when the (dynamically loaded) fieldname does not exist as attribute in the Component.

But I created an `updated(self, settings_name: str, value: str)` method of the Component, where simply every settings_name(=unicorn model) is caught at the ajax POST and processed, meaning saved into the database.

This is a clear "abuse" of Unicorn's API - I just wanted to ask you if you see a problem here, and this could maybe be used as well in django-forms-dynamic - with a custom component that has an `updated` method that does exactly this here.

What do you think?

**Reply by adamghill** [reply]
I assume you mean you are using https://www.django-unicorn.com/docs/advanced/#updated-name-value? If so, I don't think it's an abuse of the API, I think it's an ingenious way to solve your problem! 😆 I've never tried to make a dynamic form like you are doing, but I'm pleased that it all seems to work as expected.

I am still curious to dig into `django-forms-dynamic`, but based on this I hope it might "just work". I'll update this discussion if I find time to check it out.

**Reply by nerdoc** [reply]
I did it even cooler. I saw that updating the values frontend->Unicorn/updated() -> save in DB works well in, but the other way round is a problem: getting the values to the form with data... So I did it in another way: I created the properties on-the-fly in the View's mount() method with something like (simplified):

```python
class PreferencesView(FormView,UnicornView)
    def mount():
        for item in ScopedSettings.objects.all():
            setattr(PreferencesView, f"{item.namespace}__{item.key}__{item.scope.name}", item.value)
```

Now the class attributes are created "dynamically" - the `mount()` method is early enough in the process, this works **perfectly**.
I can share the code when it's a bit better ;-) a little bit spaghetti parts at the moment...

The form creates the fields dynamically too and puts a `field.widget.attrs["unicorn:model.lazy"] = dynamically_created_fieldname` into its' `__init__()` method.

Works extremely cool. But I went away from crispy, as I needed more customisation, and render the form by a normal template.

**Reply by saroipardavila** [reply]
Hi, I would love to see the final code. I'm trying to do the same thing but I'm not sure how to get everything working at the moment. Thanks a lot !
